src/sfloader/fluid_sfont.c

```diff
@@ -700,13 +700,40 @@ int fluid_sample_set_pitch(fluid_sample_t *sample, int root_key, int fine_tune)
  */
 int fluid_sample_validate(fluid_sample_t *sample, unsigned int buffer_size)
 {
+#define EXCLUSIVE_FLAGS (FLUID_SAMPLETYPE_MONO | FLUID_SAMPLETYPE_RIGHT | FLUID_SAMPLETYPE_LEFT)
+    static const unsigned int supported_flags = EXCLUSIVE_FLAGS | FLUID_SAMPLETYPE_LINKED | FLUID_SAMPLETYPE_OGG_VORBIS | FLUID_SAMPLETYPE_ROM;
+
     /* ROM samples are unusable for us by definition */
     if(sample->sampletype & FLUID_SAMPLETYPE_ROM)
     {
         FLUID_LOG(FLUID_WARN, "Sample '%s': ROM sample ignored", sample->name);
         return FLUID_FAILED;
     }
 
+    if(sample->sampletype & ~supported_flags)
+    {
+        FLUID_LOG(FLUID_WARN, "Sample '%s' has unknown flags, which might indicate that it uses a compression that it not supported by fluidsynth", sample->name);
+        return FLUID_FAILED;
+    }
+
+    if((sample->sampletype & EXCLUSIVE_FLAGS) & ((sample->sampletype & EXCLUSIVE_FLAGS) - 1))
+    {
+        FLUID_LOG(FLUID_WARN, "Sample '%s' must be either mono or left or right", sample->name);
+        return FLUID_FAILED;
+    }
+
+    if((sample->sampletype & FLUID_SAMPLETYPE_LINKED) && (sample->sampletype & EXCLUSIVE_FLAGS))
+    {
+        FLUID_LOG(FLUID_WARN, "Linked sample '%s' cannot be mono, left or right at the same time", sample->name);
+        return FLUID_FAILED;
```

Comment:
Same here: failure necessary?

Reply:
I'm open as well. But should we clear the LINKED flag, or force it to MONO or do nothing?

Reply:
I guess same as above: leave untouched if not a problem, otherwise force mono?